controllers/cluster_controller.go

```diff
@@ -262,9 +270,135 @@ func (r *ClusterReconciler) postClusterProviderReconcile(ctx context.Context, lo
+// controller.
+type PackagesClient interface {
+	EnableFullLifecycle(context.Context, logr.Logger, string, string, *v1alpha1.Image, *registrymirror.RegistryMirror, ...curatedpackages.PackageControllerClientOpt) error
+	ReconcileDelete(context.Context, logr.Logger, client.Client, *anywherev1.Cluster) error
+}
+
+func (r *ClusterReconciler) postReconcilePackagesForWorkloadCluster(ctx context.Context, log logr.Logger, cluster *anywherev1.Cluster) (err error) {
+	image, err := r.getBundleFromCluster(ctx, cluster)
+	if err != nil {
+		return err
+	}
+
+	kubeConfig, err := r.writeKubeConfigSecret(ctx, cluster)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err := kubeConfig.Remove(); err != nil {
+			log.Error(err, "removing kubeconfig file", "filename", kubeConfig.Name())
+		}
+	}()
+
+	registry := registrymirror.FromCluster(cluster)
+
+	if err := r.packagesClient.EnableFullLifecycle(ctx, log, cluster.Name, kubeConfig.Name(), image, registry); err != nil {
+		return fmt.Errorf("packages client error: %w", err)
+	}
+
+	log.V(6).Info("Installed curated packages on workload cluster", "cluster", cluster.Name)
+
+	return nil
+}
+
+// getBundleFromCluster based on the cluster's k8s version.
+func (r *ClusterReconciler) getBundleFromCluster(ctx context.Context, cluster *anywherev1.Cluster) (*v1alpha1.Image, error) {
+	bundles := &v1alpha1.Bundles{}
+	nn := types.NamespacedName{
+		Name:      cluster.Spec.BundlesRef.Name,
+		Namespace: cluster.Spec.BundlesRef.Namespace,
+	}
+	if err := r.client.Get(ctx, nn, bundles); err != nil {
+		return nil, fmt.Errorf("retrieving bundle: %w", err)
+	}
+
+	verBundle, err := r.findMatchingBundle(bundles, string(cluster.Spec.KubernetesVersion))
+	if err != nil {
+		return nil, err
+	}
+
+	return &verBundle.PackageController.HelmChart, nil
+}
+
+func (r *ClusterReconciler) findMatchingBundle(bundles *v1alpha1.Bundles, kubeVersion string) (*v1alpha1.VersionsBundle, error) {
+	var verBundle *v1alpha1.VersionsBundle
+	for _, b := range bundles.Spec.VersionsBundles {
+		if b.KubeVersion == string(kubeVersion) {
+			verBundle = &b
+			break
+		}
+	}
+	if verBundle == nil {
+		return nil, fmt.Errorf("no bundle for kube version %q", kubeVersion)
+	}
+
+	return verBundle, nil
+}
+
+// writeKubeConfigSecret so that the helm executable can pick it up to install
+// curated packages helm charts.
+func (r *ClusterReconciler) writeKubeConfigSecret(ctx context.Context, cluster *anywherev1.Cluster) (kubeconfigFile, error) {
+	kubeConfigSecret := &corev1.Secret{}
+	kubeConfigNN := types.NamespacedName{
+		Namespace: constants.EksaSystemNamespace,
+		Name:      cluster.Name + "-kubeconfig",
+	}
+	if err := r.client.Get(ctx, kubeConfigNN, kubeConfigSecret); err != nil {
+		return nil, fmt.Errorf("getting kubeconfig secret: %w", err)
+	}
+	secretBytes, err := yaml.Marshal(kubeConfigSecret)
+	if err != nil {
+		return nil, fmt.Errorf("marshaling secret %w", err)
+	}
+	f, err := os.CreateTemp("", "kubeconfig-*.yaml")
```

Comment:
Why is this necessary?

Reply:
> Why is this necessary?

Not sure what you're asking, but I'll explain what's going on...

This is pretty standard code for marshaling (or serializing if you prefer) a Go struct to YAML text, and then writing it to a file on disk. The Go struct in question is kubeconfig data that I need to write to the disk so that the helm executable can read it.

I don't know of a better/simpler way to get that data to the helm executable.

However, as per [one of your earlier comments](https://github.com/aws/eks-anywhere/pull/4970#discussion_r1123495122), this code might be removed soon. I'll be testing that possibility ASAP.

Reply:
Monday morning followup. Wow. SUPER-MEGA-ULTRA-Brain fart on my part. I see what you're asking now. 

I must not have had my head on right when I wrote this.

I'm guessing you were confused, because I took what was already just YAML text, not a Go struct, and double serialized it then wrote it to disk, and how could that possibly work?

On the bright side, this means I can almost 99% certain just get rid of this.

However, the helm executable interface still needs a file to pass on the command line. So I'll re-work the ChartInstaller implementation to skip passing the kubeconfig flag if the filename is the empty string.

Reply:
Tests indicate that indeed, this isn't necessary at all. Will remove.